dotnet/src/IntegrationTests/Agents/MixedAgentTests.cs

```diff
@@ -0,0 +1,162 @@
+            new(azureOpenAISettings.ApiKey, azureOpenAISettings.Endpoint),
+            azureOpenAISettings.ChatDeploymentName!);
+    }
+
+    private async Task ExecuteAgentAsync(
+        Kernel chatCompletionKernel,
+        OpenAIAssistantConfiguration config,
+        string modelName)
+    {
+        // Arrange
+        KernelPlugin plugin = KernelPluginFactory.CreateFromType<MenuPlugin>();
+
+        // Configure chat agent with the plugin.
+        ChatCompletionAgent chatAgent =
+            new()
+            {
+                Kernel = chatCompletionKernel,
+                Instructions = "Answer questions about the menu.",
+                Arguments = new(new OpenAIPromptExecutionSettings() { ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions }),
+            };
+        chatAgent.Kernel.Plugins.Add(plugin);
+
+        // Assistant doesn't need plug-in since it has access to the shared function result.
+        OpenAIAssistantAgent assistantAgent =
+            await OpenAIAssistantAgent.CreateAsync(
+                kernel: new(),
+                config,
+                new()
+                {
+                    Instructions = "Answer questions about the menu.",
+                    ModelId = modelName,
+                });
+
+        // Act
+        try
+        {
+            AgentGroupChat chat = new(chatAgent, assistantAgent);
+            await this.InvokeAgentAsync(chat, chatAgent, "What is the special soup?", "Clam Chowder");
+            await this.InvokeAgentAsync(chat, assistantAgent, "What is the special drink?", "Chai Tea");
+        }
+        finally
+        {
+            await assistantAgent.DeleteAsync();
+        }
+    }
+
+    private async Task InvokeAgentAsync(AgentGroupChat chat, Agent agent, string input, string expected)
+    {
+        chat.AddChatMessage(new ChatMessageContent(AuthorRole.User, input));
+
+        // Act
+        StringBuilder builder = new();
+        await foreach (var message in chat.InvokeAsync(agent))
+        {
+            builder.Append(message.Content);
+        }
+
+        // Assert
+        Assert.Contains(expected, builder.ToString(), StringComparison.OrdinalIgnoreCase);
+    }
+
+    private Kernel CreateChatCompletionKernel(AzureOpenAIConfiguration configuration)
+    {
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+            configuration.ChatDeploymentName!,
+            configuration.Endpoint,
+            configuration.ApiKey);
+
+        return kernelBuilder.Build();
+    }
+
+    private Kernel CreateChatCompletionKernel(OpenAIConfiguration configuration)
+    {
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+
+        kernelBuilder.AddOpenAIChatCompletion(
+            configuration.ModelId!,
+            configuration.ApiKey);
+
+        return kernelBuilder.Build();
+    }
+
+    public sealed class MenuPlugin
```

Comment:
If the scope of this plugin is just this test class, make it private.

```suggestion
    private sealed class MenuPlugin
```